### PR TITLE
Pb 1831 footnote issue refactor

### PR DESCRIPTION
**Pull request by ricardopressbooks**
https://github.com/pressbooks/pressbooks/issues/1831

## How to test
- Create a footnote in a chapter with different elements like lists, images, quotes, etc
- Create more than 1 footnote in the same paragraph

Check in web and PDF export footnotes have the expected behavior.

### CODE REVIEW

File: assets/src/scripts/export-footnotes.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+function moveFootnotes() {
+	var footnotes = document.getElementsByClassName( 'footnote-indirect' );
```
Comment (cagp-dev-mtl) [inline assets/src/scripts/export-footnotes.js:2]: Pls change all `var` for `let`

File: inc/modules/export/prince/class-pdf.php [hunk 2]
```diff
@@ -119,6 +120,7 @@ function convert() {
 		// Save PDF as file in exports folder
 
 		$prince = new \PrinceXMLPhp\PrinceWrapper( PB_PRINCE_COMMAND );
+
```
Comment (cagp-dev-mtl) [inline inc/modules/export/prince/class-pdf.php:123]: remove trailing new line.

File: inc/modules/export/xhtml/class-xhtml11.php [hunk 3]
```diff
@@ -520,6 +529,32 @@ function doEndnotes( $id ) {
...
+	 * Content for footnotes.
+	 *
+	 * @see footnoteShortCode
+	 *
+	 * @param $id
+	 *
+	 * @return string
+	 */
+	function doFootnotes( $id ) {
+
```
Comment (cagp-dev-mtl) [inline inc/modules/export/xhtml/class-xhtml11.php:542]: trailing newline (remove)

File: inc/modules/export/xhtml/class-xhtml11.php [hunk 4]
```diff
@@ -353,14 +359,14 @@ public function transformGenerator() : \Generator {
...
 		}
-
 		echo "</head>\n<body lang='{$this->lang}' ";
 		if ( ! empty( $_GET['optimize-for-print'] ) ) {
 			echo "class='print' ";
 		}
 		echo ">\n";
 		$replace_token = uniqid( 'PB_REPLACE_INNER_HTML_', true );
 		echo $replace_token;
+
```
Comment (cagp-dev-mtl) [inline inc/modules/export/xhtml/class-xhtml11.php:369]: Trailing newline (Remove)

File: inc/modules/export/prince/class-pdf.php [hunk 5]
```diff
@@ -136,6 +138,12 @@ function convert() {
 		}
 
 		$prince->addStyleSheet( $css_file );
+
+		// Footnote hack to allow more complex DOM
```
Comment (cagp-dev-mtl) [inline inc/modules/export/prince/class-pdf.php:142]: pls remove comment

**Comment by cagp-dev-mtl** [comment]
In the future try to use jQuery, e.g. `export.js`

`var footnotes = document.getElementsByClassName( 'footnote-indirect' );`

becomes

`let $footnotes = $( '.footnote-indirect' );`

etc

if you are looping a jQuery object you can use:

`$footnotes.each( function() { .................. } )`

**Comment by ricardopressbooks** [comment]
@cagp-dev-mtl thank you for your feedback.
I did some fixes according to your requests.

About JS Script I tried before to use jQuery with this script (instead the current):
```js
function moveFootnotes() {
	let footnotes = $( '.footnote-indirect' );
	for ( let i = 0; i < footnotes.length; i++ ) {
		let ref = $('#' + footnotes[i].attr('data-fnref'));
		if (ref) footnotes[i].append(ref);
	}
}
if ( typeof Prince != 'undefined' ) {
	moveFootnotes();
}
```

But seems to be jQuery is not available for Prince in our current configuration. I guess doesn't makes sense for this small and simple script, what do you think?

**Comment by SteelWagstaff** [comment]
May be helpful? https://www.princexml.com/doc/javascript/ + https://www.princexml.com/doc/prince-input/#applying-javascript-in-prince

**Comment by ricardopressbooks** [comment]
> May be helpful? https://www.princexml.com/doc/javascript/ + https://www.princexml.com/doc/prince-input/#applying-javascript-in-prince

Yes I actually used that approach for this solution.

**Comment by ricardopressbooks** [comment]
@cagp-dev-mtl thank you for the feedback.
I tried with both (using `$` and `jQuery`) and doesn't work. This is my logs locally trying to use for example jQuery (applying same script you shared):
```
Mon Jul 13 16:51:54 2020: ---- begin
Mon Jul 13 16:51:54 2020: https://pressbooks.test/app/plugins/pressbooks/assets/dist/scripts/export-footnotes.js?id=ef3dd8e81c30fbc37b41:1: error: ReferenceError: no such property: jQuery
Mon Jul 13 16:51:54 2020: finished: success
Mon Jul 13 16:51:54 2020: ---- end
```
I fixed the syntax issue and I applied brackets correctly.

**Comment by ricardopressbooks** [comment]
@SteelWagstaff could you test if it works for Prince in some network? Otherwise we should revert this PR as well according we talked with @cagp-dev-mtl during our planning meeting.

**Pull request merged**